src/applications/Utilities/src/Utilities.cxx

```diff
@@ -621,6 +631,63 @@ int algorithmsRunner()
     inputImage->TransformPhysicalPointToIndex(indexToConvert, output);
     std::cout << indexToConvert << " ==> " << output << "\n";
   }
+  else if (requestedAlgorithm == LabelSimilarity)
+  {
+    // this filter only works on unsigned int type
+    using DefaultImageType = itk::Image< unsigned int, TImageType::ImageDimension >;
+    auto inputImage = cbica::ReadImage< DefaultImageType >(inputImageFile);
+    auto referenceImage = cbica::ReadImage< DefaultImageType >(referenceMaskForSimilarity);
+    auto uniqueLabels = cbica::GetUniqueValuesInImage< DefaultImageType >(inputImage);
+    auto uniqueLabelsRef = cbica::GetUniqueValuesInImage< DefaultImageType >(referenceImage);
+
+    // sanity check
+    if (uniqueLabels.size() != uniqueLabelsRef.size())
+    {
+      std::cerr << "The number of unique labels in input and reference image are not consistent.\n";
+      return EXIT_FAILURE;
+    }
+    else
+    {
+      for (size_t i = 0; i < uniqueLabels.size(); i++)
+      {
+        if (uniqueLabels[i] != uniqueLabelsRef[i])
+        {
+          std::cerr << "The label values in input and reference image are not consistent.\n";
+          return EXIT_FAILURE;
+        }
+      }
+    }
+
+    auto similarityFilter = itk::LabelOverlapMeasuresImageFilter< DefaultImageType >::New();
+
+    similarityFilter->SetSourceImage(inputImage);
+    similarityFilter->SetTargetImage(referenceImage);
+    similarityFilter->Update();
+
+    std::cout << "=== All Labels ===\n";
+    std::cout << "Total Overlap:   " << similarityFilter->GetTotalOverlap() << "\n";
+    std::cout << "Union (Jaccard): " << similarityFilter->GetUnionOverlap() << "\n";
+    std::cout << "Mean (DICE):     " << similarityFilter->GetMeanOverlap() << "\n";
+    std::cout << "Volume Sim.:     " << similarityFilter->GetVolumeSimilarity() << "\n";
+    std::cout << "False Neg. Err.: " << similarityFilter->GetFalseNegativeError() << "\n";
+    std::cout << "False Pos. Err.: " << similarityFilter->GetFalsePositiveError() << "\n";
+
+    if (uniqueLabels.size() > 2) // basically if there is something more than 0 and 1
+    {
+      std::cout << "=== Non-zero Labels ===\n";
```

Comment:
You might want to make the distinction between the two a bit more clear, like:
- Masked area
- Individual labels 

Also you can add an extra space in the output between the different result sets.

Reply:
Done.